nodebuilder/share/bitswap.go

```diff
@@ -58,8 +52,12 @@ func blockstoreFromDatastore(ds datastore.Batching) (blockstore.Blockstore, erro
 	return blockstore.NewBlockstore(ds), nil
 }
 
-func blockstoreFromEDSStore(store *store.Store) (blockstore.Blockstore, error) {
-	withCache, err := store.WithCache("blockstore", defaultBlockstoreCacheSize)
+func blockstoreFromEDSStore(store *store.Store, blockStoreCacheSize int) (blockstore.Blockstore, error) {
+	if blockStoreCacheSize == 0 {
+		// no cache, return plain blockstore
+		return &bitswap.Blockstore{Getter: store}, nil
+	}
+	withCache, err := store.WithCache("blockstore", blockStoreCacheSize)
```

Comment:
Also, might be a special case on WithCache that returns itself.

Reply:
Ok, CachedStore is not an interface, so won't work

Reply:
Good point. Perhaps user should not be allowed call WithCache with size <= 0. I've added the check for that